### PR TITLE
Fix to handle EADDRINUSE exception when running multiple feature serv…

### PR DESCRIPTION
**Pull request by Megamuch**
…er services on the same machine.

.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -106,6 +115,7 @@ sessionTracker.on('idle', () => {
 
 const getCount = () => sessionTracker.count;
 const healthCheck = require('@jambonz/http-health-check');
+const { on } = require('events');
```
Comment (davehorton) [inline app.js:118]: is this used?

File: app.js [hunk 2]
```diff
@@ -92,7 +93,15 @@ app.use((err, req, res, next) => {
   logger.error(err, 'burped error');
   res.status(err.status || 500).json({msg: err.message});
 });
+
 const httpServer = app.listen(PORT);
+process.on('uncaughtException', function(err) {
+    if(err.errno === 'EADDRINUSE' && useMultiple) {
```
Comment (davehorton) [inline app.js:99]: I'm not sure 'useMultiple' is needed.  It seems completely redundant with the information provided the `JAMBONES_IGNORE_EADDRINUSE` env.  Or, to put it another way, if someone configured the system with `JAMBONES_IGNORE_EADDRINUSE` isn't that enough to tell us that this new code needs to be executed?

File: app.js [hunk 3]
```diff
@@ -92,7 +93,15 @@ app.use((err, req, res, next) => {
...
+        const httpServer = app.listen(PORT++);
```
Comment (davehorton) [inline app.js:100]: We should burp out a log message here that we are now going to attempt to bind to a new port

Comment (davehorton) [inline app.js:100]: It seems to me that this error will also be thrown from lib/utils/aws-sns-lifecycle, yet here we are assuming that the bind error can only come from the healthcheck port?

**Review by davehorton — CHANGES_REQUESTED** [review]
It seems to me that

**Comment by hkrutzer** [comment]
This doesn't really seem like a responsibility for Jambonz to handle. Rather it is the job of configuration management tools. And if it is, a more robust solution like https://github.com/sindresorhus/get-port should probably be used.

File: app.js (hunk 2 above)
Comment (Megamuch) [reply app.js:99]: correct. I removed it.

File: app.js (hunk 3 above)
Comment (Megamuch) [reply app.js:100]: added.

**Comment by davehorton** [comment]
I'm addressing this in a different PR

**Pull request closed without merging**